src/FSharpPlus/Data/NonEmptyMap.fs

```diff
@@ -0,0 +1,266 @@
+    /// <exception cref="System.Collections.Generic.KeyNotFoundException">Thrown if the key does not exist in the map.</exception>
+    /// <returns>The first key for which the predicate evaluates true.</returns>
+    let findKey predicate (table: NonEmptyMap<_, _>) = Map.findKey predicate table.Value
+
+    /// <summary>Returns the key of the first mapping in the collection that satisfies the given predicate.
+    /// Returns 'None' if no such element exists.</summary>
+    /// <param name="predicate">The function to test the input elements.</param>
+    /// <param name="table">The input map.</param>
+    /// <returns>The first key for which the predicate returns true or None if the predicate evaluates to false for each key/value pair.</returns>
+    let tryFindKey predicate (table: NonEmptyMap<_, _>) = Map.tryFindKey predicate table.Value
+
+    /// <summary>The number of bindings in the map.</summary>
+    let count (table: NonEmptyMap<_, _>) = Map.count table.Value
+
+    // reduce functions
+
+    let reduce reduction (map: NonEmptyMap<_, _>) = Seq.reduce reduction (toSeq map)
+    let reduceBack reduction (map: NonEmptyMap<_, _>) = Seq.reduceBack reduction (toSeq map)
+
+    // extensions from FSharpPlus
+
+    let keys   (source: NonEmptyMap<'Key, 'T>) = Seq.map (fun (KeyValue(k, _)) -> k) source
+    let values (source: NonEmptyMap<'Key, 'T>) = Seq.map (fun (KeyValue(_, v)) -> v) source
+
+    /// <summary>Map values of the original Map.</summary>
+    /// <remarks>Keys remain unchanged.</remarks>
+    /// <param name="f">The mapping function.</param>
+    /// <param name="x">The input Map.</param>
+    ///
+    /// <returns>The mapped Map.</returns>
+    let mapValues f (x: NonEmptyMap<'Key, 'T>) = map (fun _ -> f) x
+
+    let iterValues f (x: NonEmptyMap<_, 'T>) = iter (fun _ -> f) x
+
+    let unzip (source: NonEmptyMap<'Key, 'T1 * 'T2>) = mapValues fst source, mapValues snd source
+
+    /// Returns the union of two maps, using the combiner function for duplicate keys.
+    let unionWith combiner (source1: NonEmptyMap<'Key, 'Value>) (source2: NonEmptyMap<'Key, 'Value>) =
+        fold (fun m k v' -> add k (match tryFind k m with Some v -> combiner v v' | None -> v') m) source1 source2
+
+    /// Returns the union of two maps, preferring values from the first in case of duplicate keys.
+    let union (source: NonEmptyMap<'Key, 'T>) (altSource: NonEmptyMap<'Key, 'T>) = unionWith (fun x _ -> x) source altSource
+
+#if !FABLE_COMPILER
+    let inline traverse (f: 'T->'``Functor<'U>``) (m: NonEmptyMap<'K, 'T>) : '``Functor<NonEmptyMap<'K, 'U>>`` =
+        let m' = traverse f (toMap m) : '``Functor<Map<'K, 'U>>``
+        ofMap <!> m' : '``Functor<NonEmptyMap<'K, 'U>>``
+
+    let inline traversei (f: 'K -> 'T -> '``Functor<'U>``) (m: NonEmptyMap<'K, 'T>) : '``Functor<NonEmptyMap<'K, 'U>>`` =
+        let m' = traversei f (toMap m) : '``Functor<Map<'K, 'U>>``
+        ofMap <!> m' : '``Functor<NonEmptyMap<'K, 'U>>``
```

Comment:
if we look at the traversei implementation for `Map`, it seems like the return type does not need to be Map?

Reply:
For instance if I look at 
https://github.com/fsprojects/FSharpPlus/blob/1c5846a51a5083b94c12090e727ad6743b29c629/tests/FSharpPlus.Tests/General.fs#L1112-L1115

Reply:
Though it looks like you have wrap the result in option here, but it's the same type of structure inside